website/server/models/group.js

```diff
@@ -1597,7 +1593,16 @@ schema.methods.syncTask = async function groupSyncTask (taskToSync, user) {
   // add tag if missing
   if (matchingTask.tags.indexOf(group._id) === -1) matchingTask.tags.push(group._id);
 
-  toSave.push(matchingTask.save(), taskToSync.save(), user.save());
+  toSave.push(matchingTask.save(), user.save());
+  return Promise.all(toSave);
+};
+
+schema.methods.linkTask = async function groupLinkTask (linkingTask, user) {
+  const toSave = [];
+  if (linkingTask.group.assignedUsers.indexOf(user._id) === -1) {
+    linkingTask.group.assignedUsers.push(user._id);
+  }
+  toSave.push(linkingTask.save(), this.syncTask(linkingTask, user));
```

Comment:
syncTask changes `linkingTask` but doesn't save it, it's saved while `syncTask ` runs so it might not catch the changes made there?